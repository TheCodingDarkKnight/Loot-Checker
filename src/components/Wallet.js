import React, { Component } from "react";
import { connect } from "react-redux";

export class Wallet extends Component {

    render() {
        console.log(this.props)
        return (<h3 className="balance">Wallet balance: {this.props.balance}</h3>)
    }
}

export default connect(state => ({ balance: state.balance }), undefined)(Wallet);