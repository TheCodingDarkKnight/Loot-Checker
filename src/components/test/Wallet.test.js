import React from "react";
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import { Wallet } from "../Wallet";

configure({ adapter: new Adapter() });

describe("Wallet", () => {
    const props = { balance: 20 }
    const wallet = shallow(<Wallet {...props} />);

    it("Renders properly", () => {
        expect(wallet).toMatchSnapshot();
    });

    it("Displays the balance from props", () => {
        expect(wallet.find(".balance").text()).toEqual(`Wallet balance: ${props.balance}`)
    })
})