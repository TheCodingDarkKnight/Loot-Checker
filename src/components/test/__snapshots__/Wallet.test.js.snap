// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet Renders properly 1`] = `ShallowWrapper {}`;
